### ISSUE TITLE
Traefik fails to acquire Lets Encrypt certificate for NS1 subdomain

### ISSUE THREAD

**Issue by 0xFelix** [containous/traefik#3871]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I configured NS1 as dnsChallenge provider and added a frontend rule for a Host that is in a subdomain that is delegated to NS1. (myhost.sub.mydomain.tld)

### What did you expect to see?

I expected a Lets Encrypt certificate to be created.

### What did you see instead?

The creation fails with:

```
Unable to obtain ACME certificate for domains \"myhost.sub.mydomain.tld\" detected thanks to rule \"Host:myhost.sub.mydomain.tld\" : cannot obtain certificates: acme: Error -> One or more domains had a problem:\n[myhost.sub.mydomain.tld] Error presenting token: zone does not exist\n
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.6.6
Codename:     tetedemoine
Go version:   go1.10.3
Built:        2018-08-20_01:10:06PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
debug = false

logLevel = "DEBUG"
defaultEntryPoints = ["https","http"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]

[retry]

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "sub.mydomain.tld"
watch = true
exposedByDefault = false

[acme]
email = "me@mydomain.tld"
storage = "acme.json"
entryPoint = "https"
onHostRule = true
caServer = "https://acme-staging-v02.api.letsencrypt.org/directory"
  [acme.dnsChallenge]
  provider = "ns1"
  delayBeforeCheck = 0

[web]
address = ":8080"
```

I guess this has something to do with commit https://github.com/xenolf/lego/commit/771679fe3df672ef4c05efcd4b88c40c44d4b1e2. Did this commit already land in traefik? As far as I understood traefik uses LEGO.

**Comment by ldez** [comment]
@0xFelix Thanks for reporting :+1: 

The PR https://github.com/xenolf/lego/pull/463 is not currently use by v1.6.6.
I will make a PR to update Lego soon.

**Comment by ldez** [comment]
@0xFelix If I create a custom Traefik image or a custom Lego version, can you do some testing for me?

**Comment by 0xFelix** [comment]
@ldez Yeah I could do that

**Comment by ldez** [comment]
@0xFelix great, could you try `ldez/traefik:ns1` docker image?

**Comment by traefiker** [comment]
Closed by #3915.

**Comment by 0xFelix** [comment]
@ldez I tried your image, it does work! 👍
